src/slices/AccountSlice.ts

```
@@ -142,11 +196,119 @@ export const getBalances = createAsyncThunk(
+    }
+
+    return {
+      giving: {
+        sohmGive: +giveAllowance,
+        donationInfo: donationInfo,
+        loading: false,
+      },
+    };
+  },
+);
+
+/**
+ * Provides the details of deposits/donations provided by a specific wallet.
+ *
+ * This differs from the standard `getDonationBalances` function because it uses a alternative
+ * sOHM contract that allows for manual rebases, which is helpful during testing of the 'Give' functionality.
+ */
+export const getMockDonationBalances = createAsyncThunk(
+  "account/getMockDonationBalances",
+  async ({ address, networkID, provider }: IBaseAddressAsyncThunk) => {
+    let giveAllowance = 0;
+    let donationInfo: IUserDonationInfo = {};
+
+    if (addresses[networkID] && addresses[networkID].MOCK_SOHM) {
+      const mockSohmContract = new ethers.Contract(addresses[networkID].MOCK_SOHM as string, MockSohm, provider);
+      giveAllowance = await mockSohmContract._allowedValue(address, addresses[networkID].MOCK_GIVING_ADDRESS);
+      const givingContract = new ethers.Contract(
+        addresses[networkID].MOCK_GIVING_ADDRESS as string,
+        OlympusMockGiving,
+        provider,
+      );
+
+      try {
+        // NOTE: The BigNumber here is from ethers, and is a different implementation of BigNumber used in the rest of the frontend. For that reason, we convert to string in the interim.
+        let allDeposits: [string[], BigNumber[]] = await givingContract.getAllDeposits(address);
+        for (let i = 0; i < allDeposits[0].length; i++) {
+          if (allDeposits[1][i] !== BigNumber.from(0)) {
+            // Store as a formatted string
+            donationInfo[allDeposits[0][i]] = ethers.utils.formatUnits(allDeposits[1][i], "gwei");
+          }
+        }
+      } catch (e: unknown) {
+        console.error(e);
+      }
+    } else {
+      console.error("Unable to find MOCK_SOHM contract on chain ID " + networkID);
```

should we hide this error on mainnet?